js/liveblog.js

```diff
@@ -382,6 +398,147 @@ window.liveblog = window.liveblog || {};
+		liveblog.check_notification_status('load');
+
+		// Watch enable checkbox for any change
+		liveblog.$checkbox_enable.on( 'change', function() {
+			if ( this.checked ) {
+				liveblog.check_notification_status('checked');
+			} else {
+				liveblog.check_notification_status('unchecked');
+			}
+		} );
+	};
+
+	liveblog.set_notification_status = function( status ) {
+		var checked, animation;
+
+		if ( status === 'granted' ) {
+			checked = true;
+		} else if ( status === 'denied' ) {
+			checked = false;
+		}
+
+		liveblog.$checkbox_enable.attr('checked', checked);
+		localStorage.setItem('liveblog-notifications', checked);
+	};
+
+	liveblog.check_notification_status = function( event ) {
+		if ( event === 'load' ) {
+
+			// Handle if notifications permissions are not granted on load
+			if ( Notification.permission !== 'granted' ) {
+				liveblog.set_notification_status('denied');
+
+			// If notifications are are granted and user enabled
+			} else if ( Notification.permission === 'granted' && liveblog.parse_local_storage('liveblog-notifications') ) {
+				liveblog.set_notification_status('granted');
+			}
+
+		} else if ( event === 'checked' ) {
+			if ( Notification.permission === 'granted' ) {
+				liveblog.set_notification_status('granted');
+
+			} else if ( Notification.permission === 'denied' ) {
+				liveblog.set_notification_status('denied');
+				alert(liveblog_settings.notification_blocked_message);
+
+			} else {
+				Notification.requestPermission(function (permission) {
+					if ( permission === 'granted' ) {
+						liveblog.set_notification_status('granted');
+					} else {
+						liveblog.set_notification_status('denied');
+					}
+				});
+			}
+
+		} else if ( event === 'unchecked' ) {
+			liveblog.set_notification_status('denied');
+		}
+	};
+
+	liveblog.notify = function( $new_entry ) {
+
+		/*
+		1. Make sure admins don't recieve notifications
+		2. Ensure user has enabled notifications (JSON parse for bool)
+		3. Make sure the document doesn't have focus
+		*/
+		if ( liveblog_settings.is_admin || ! liveblog.parse_local_storage('liveblog-notifications') || document.hasFocus() ) {
+			return;
+		}
+
+		var notify,
+			entry_text = liveblog.get_notification_entry_text($new_entry),
+			entry_icon = liveblog.get_notification_entry_icon($new_entry),
+			notifications_enabled = liveblog.parse_local_storage('liveblog-notifications');
+
+		if ( notifications_enabled && $new_entry.hasClass(liveblog_settings.class_key) ) {
+			liveblog.spawn_notification(liveblog_settings.notification_title, {body: entry_text, icon: entry_icon});
+		}
+	};
+
+	liveblog.get_notification_entry_text = function( $entry ) {
+		var original_content, $original_content, entry_text;
+
+		// Grab original content from data-attr
+		original_content = $entry.find('.liveblog-entry-text').data('original-content');
+
+		// Convert to jQuery object and wrap it so we can manipulate it
+		$original_content = $('<div/>').html(original_content);
```

Comment:
This is kinda unsafe, because we can't guarantee `original_content` isn't nasty. See https://vip.wordpress.com/2015/03/25/preventing-xss-in-javascript/